cloudinit/sources/helpers/upcloud.py

```diff
@@ -0,0 +1,223 @@
+# Author: Antti Myyrä <antti.myyra@upcloud.com>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import json
+
+from cloudinit import dmi
+from cloudinit import log as logging
+from cloudinit import net as cloudnet
+from cloudinit import url_helper
+
+LOG = logging.getLogger(__name__)
+
+
+def convert_to_network_config_v1(config):
```

Comment:
Hi @ajmyyra, thanks for this submission! We're about to take a closer look at this, but one thing I noticed up-front is that you're generating [v1](https://cloudinit.readthedocs.io/en/latest/topics/network-config-format-v1.html#network-config-v1) network configuration. Is there a particular reason that you chose generating that over [v2](https://cloudinit.readthedocs.io/en/latest/topics/network-config-format-v2.html#network-config-v2) config?

(v2 is, of course, a more recent iteration of our network config, so is preferable from that point of view. It also has the advantage that it can be passed through to the filesystem on netplan-enabled systems (i.e. modern Ubuntu), without further processing internal to cloud-init: v1 always has to be converted to a target format.)

Reply:
Hi! Reason for this was that I wasn't aware if v2 is already fully supported on non-netplan distros, so decided to play it safe. I can quite easily convert it to v2 before this is merged if that's desired.